dhis-2/dhis-api/src/main/java/org/hisp/dhis/attribute/Attribute.java

```diff
@@ -258,6 +260,18 @@ public void setAttribute( ObjectType type, boolean isAttribute )
         }
     }
 
+    @JsonIgnore
+    public Set<String> getObjectTypes()
+    {
+        return objectTypes.stream().map( ObjectType::name ).collect( toSet() );
+    }
+
+    public void setObjectTypes( Set<String> objectTypes )
+    {
+        this.objectTypes.clear();
```

Comment:
Is there some theoretical race condition potential here? I.e. two threads calling this non-synchronized method might override/clear the set.

Reply:
The modified collection is not one that hibernate is aware of as we convert back and forth but you are right, this is a race condition should multiple threads use the same Attribute instance. Will think of a better way to do this.